relayer/submit_transaction.go

```diff
@@ -57,6 +61,10 @@ func ValidateAndSendTransaction(
 		return nil, err
 	}
 
+	if err = checkForMinClaimBalance(ctx, ec); err != nil {
+		return nil, err
+	}
+
 	// Lock the wallet's nonce until we get a receipt
```

Comment:
use the same gas price in this function in `txOpts` below, otherwise it could fail still

Reply:
`txOpts.GasPrice = suggestedGasPrice`

Reply:
Unfortunately, it won't fail. (What I discovered before adding this function.) It it failed when creating the transaction, that would be awesome, but unless you don't have the base fee (which you probably do) it will create a transaction that might not get mined quickly or at all.

That said, I implemented the suggestion in this commit:
https://github.com/AthanorLabs/atomic-swap/pull/366/commits/d23664fc7030078cf637a33d0ebe71905f679146